Add support for netstandard2.0 TFM

am11 opened the pull request:
* Update projects to CPS based csprojs.
* Add explicit support for .NET Standard 2.0
* Update AppVeyor and TravisCI scripts
  * Add macOS build job

am11 commented:
@neris, for some reason, the decryption tool is requiring a passphrase:

> Error: Secret passphrase is missing

I thought it will work under your account right away. For my testing, I commented out these lines in my fork and uncommented before issuing the PR. :)

am11 commented on .travis.yml:
```diff
@@ -4,50 +4,62 @@
-  - dotnet build --framework "net45"
-  - dotnet build --framework ".NETPortable,Version=v4.0,Profile=Profile136"
-  - dotnet build --framework ".NETPortable,Version=v4.0,Profile=Profile328"
-  - dotnet build --framework ".NETPortable,Version=v4.5,Profile=Profile259"
```
Removed Profile259, as this is equivalent to netstandard1.0, as per the NuGet TFM docs https://docs.microsoft.com/en-us/nuget/schema/target-frameworks

neris commented:
Thank you very much for the contribution.
If looks very good, but so far I can see a couple of problems: 

1. We don't need to compile the library against the `netstandard2.0`. We're not using any features from it and `netstandard1.3` is fully compatible with .NET Core 2.0. We do need `netcoreapp2.0` target for test projects though.

2. I can't build the project with the `dotnet` CLI command for `net20` and `net35` (although it builds just fine from Visual Studio). Error message:
   ```
   error MSB3644: The reference assemblies for framework ".NETFramework,Version=v3.5" were not found. To resolve this, install the SDK or Targeting Pack for this framework version or retarget your application to a version of the framework for which you have the SDK or Targeting Pack installed. Note that assemblies will be resolved from the Global Assembly Cache (GAC) and will be used in place of reference assemblies. Therefore your assembly may not be correctly targeted for the framework you intend.
   ```
   It seems like it's because of this issue: https://github.com/Microsoft/msbuild/issues/1333
   As a possible workaround we can add this option to the csproj file:
   ```
   <FrameworkPathOverride Condition="'$(TargetFramework)' == 'net35' OR '$(TargetFramework)' == 'net20'">$(MSBuildProgramFiles32)\Reference Assemblies\Microsoft\Framework\.NETFramework\v3.5\Profile\Client</FrameworkPathOverride>
   ```
   But it's not the prettiest solution.

am11 commented:
@neris, both the AppVeyor and TravisCI builds+tests passed in my fork before I uncommented the `secret` line e.g.: https://ci.appveyor.com/project/am11/ngettext/build/1.0.26 and https://travis-ci.org/am11/NGettext/jobs/266319943. Can you please unblock the secure-string passpharse issue in CI, so we can see some green builds?

> We don't need to compile the library against the netstandard2.0. We're not using any features from it and netstandard1.3 is fully compatible with .NET Core 2.0. We do need netcoreapp2.0 target for test projects though.

I totally understand the rationale that the lower TFM targets the higher ones. However, the good packages tend to explicitly target each TFM for a slight differences to provide the best experience possible. One change here is that netstandard1.3 requires `System.Diagnostics.TraceSource` dependency, while netstandard2.0 doesn't. 

> FrameworkPathOverride 

As you can see I have this environment variable set in .travis.yml for .NET Framework monikers (and `TargetFrameworkRootPath` for PCL monikers), but on Windows, it wasn't required on AppVeyor. Does it fix the issue in VS2017 by setting this property in both csprojs under src directory? If so, I will set it by default for Windows, and TravisCI will continue to override it from environment variable.

neris commented:
> Can you please unblock the secure-string passpharse issue in CI, so we can see some green builds?
AppVeyor and TravisCI don't expose secret variables to PR builds (security restriction). You can remove the signing directives. Since the project will use MSBuild, I'll be able to set the signing properties from 

CLI when publishing the package (I couldn't do so with old xproj system).

> However, the good packages tend to explicitly target each TFM for a slight differences to provide the best experience possible. One change here is that netstandard1.3 requires System.Diagnostics.TraceSource dependency, while netstandard2.0 doesn't.

Makes sense, you can leave it there.

> but on Windows, it wasn't required on AppVeyor

Because it executes MSBuild directly, without the `dotnet build` command. Hopefully Microsoft will fix the issue. Until then, I think it's better to add the default `FrameworkPathOverride` option to the project file.

am11 commented:
@neris, I have removed the signing directives from both CI scripts and added the missing override properties for net20 and net35 TFMs. All builds are green now. :)

neris commented:
I can still see a couple of minor problems, but I'll fix them myself before merging into master.
@am11, thank you very much for your work.
